File: src/Budgeteer.App/Client/ts/Pages/Auth/LoginResultModel.ts
import { LoginResultCode } from './LoginResultCode';

/**
 * Modelliert das Ergebnis eines Login-Prozesses.
 */
export class LoginResultModel {
  /**
   * Initialisiert eine neue Instanz der Klasse.
   *
   * @param copy Die ggf. zu kopierenden Daten.
   */
  constructor(copy?: Partial<LoginResultModel>) {
    this.code = copy?.code ?? LoginResultCode.invalidData;
  }

  /**
   * Statische Instanz, die einen Fehlercode für einen unbekannten Fehler enthält.
   */
  public static readonly Failure = new LoginResultModel({ code: LoginResultCode.invalidData });

  /**
   * Der Ergebniscode des Logins.
   */
  public readonly code: LoginResultCode;
}

File: src/Budgeteer.App/Logic/Api/Auth/LoginResultModel.cs
﻿// -----------------------------------------------------------------------------------------------------------------------------------------
// <copyright file="LoginResultModel.cs" company="Lukas Tietze">
// Copyright (c) Lukas Tietze. All rights reserved.
// </copyright>
// -----------------------------------------------------------------------------------------------------------------------------------------

namespace Budgeteer.App.Logic.Api.Auth;

/// <summary>
/// Modelliert das Ergebnis eines Login-Prozesses.
/// </summary>
public class LoginResultModel
{
    /// <summary>
    /// Holt oder setzt den Ergebniscode des Logins.
    /// </summary>
    public LoginResultCode Code { get; set; }
}
